pkg/private/build_tar.py

```diff
@@ -278,34 +277,32 @@ def add_tree(self, tree_top, destpath, mode=None, ids=None, names=None):
             uname=names[0],
             gname=names[1])
 
-  def add_manifest_entry(self, entry_list, file_attributes):
-    entry = manifest.ManifestEntry(*entry_list)
-
+  def add_manifest_entry(self, entry, file_attributes):
     # Use the pkg_tar mode/owner remaping as a fallback
-    non_abs_path = entry.dest.strip('/')
+    non_abs_path = entry["dest"].strip('/')
     if file_attributes:
       attrs = file_attributes(non_abs_path)
     else:
       attrs = {}
     # But any attributes from the manifest have higher precedence
-    if entry.mode is not None and entry.mode != '':
-      attrs['mode'] = int(entry.mode, 8)
-    if entry.user:
-      if entry.group:
-        attrs['names'] = (entry.user, entry.group)
+    if entry["mode"] is not None and entry["mode"] != '':
+      attrs['mode'] = int(entry["mode"], 8)
+    if entry["user"]:
+      if entry["group"]:
+        attrs['names'] = (entry["user"], entry["group"])
       else:
         # Use group that legacy tar process would assign
-        attrs['names'] = (entry.user, attrs.get('names')[1])
-    if entry.entry_type == manifest.ENTRY_IS_LINK:
-      self.add_link(entry.dest, entry.src, **attrs)
-    elif entry.entry_type == manifest.ENTRY_IS_DIR:
-      self.add_empty_dir(entry.dest, **attrs)
-    elif entry.entry_type == manifest.ENTRY_IS_TREE:
-      self.add_tree(entry.src, entry.dest, **attrs)
-    elif entry.entry_type == manifest.ENTRY_IS_EMPTY_FILE:
-      self.add_empty_file(entry.dest, **attrs)
+        attrs['names'] = (entry["user"], attrs.get('names')[1])
+    if entry["type"] == "symlink":
+      self.add_link(entry["dest"], entry["src"], **attrs)
+    elif entry["type"] == "dir":
```

Comment:
Can you keep the RHS as manifest.ENTRY_IS_DIR.  That gives us a little more compile time support for not shooting ourselves in the foot.

Reply:
> I forgot that piece of code just unpacks, consumes and then throws it away. So there really is no memory concern. I guess this is mostly fine.

I thought the memory usage concerns were within the long-lived Bazel process instead of the python scripts it runs.  Given the above, can we still proceed?  I'm happy to look into cleaning this up.

> Can you keep the RHS as manifest.ENTRY_IS_DIR. That gives us a little more compile time support for not shooting ourselves in the foot.

I'll take a look.